### ISSUE TITLE
resp not terminating route when inside template

### ISSUE THREAD

**Issue by PMunch** [dom96/jester#114]
I'm writing a fairly simple server that uses a MySQL back-end and outputs JSON. MySQL connections time out after a while so I added a try-except around it. But I wanted to do this for most my routes so I created a template which looks something like this:

```
template tryWithDbError(internal: untyped): untyped =
  try:
    internal
  except DbError as error:
    if db.errno == 2006: # Error code for timed out connection
      try:
        db = open("","","","") # Credentials redacted
        internal
      except DbError as error:
        echo "Database error while processing request($1): $2" % [$db.errno, error.msg]
        resp Http500, errorWrap "Database error while processing request($1): $2" % [$db.errno, error.msg]
    else:
      echo "Database error while processing request($1): $2" % [$db.errno, error.msg]
      resp Http500, errorWrap "Database error while processing request($1): $2" % [$db.errno, error.msg]
```

And it works great. But if I have code like this:
```
tryWithDbError:
  db.exec(sql"DROP TABLE bobbytables")
  resp Http200, "Dropped bobbytables"
resp Http404, "This part of the code should never be reached"
```
The server will respond with the Http404. It seems like Jester returns from the scope of the template when the resp is found and not the scope of the route as would be expected.

**Comment by PMunch** [comment]
As we had a discussion about this over on IRC I figured I should put that here (edited out interleaving discussions):
```
18:44:07	PMunch	Any word on this: https://github.com/dom96/jester/issues/114 +
18:46:27	dom96	Didn't get a chance to look at it yet I'm afraid
18:50:46	PMunch	Hmm, what does resp actually do under the hood?
18:53:33	bozaloshtsh	https://github.com/dom96/jester/blob/master/jester.nim#L336
18:53:47	bozaloshtsh	it doesn't return
18:54:57	PMunch	bozaloshtsh, last line # The ``route`` macro will add a 'return' after the invokation of this template.
18:55:07	dom96	https://github.com/dom96/jester/blob/master/jester.nim#L613
18:57:28	PMunch	I'm guessing the route macro looks at the AST before the template is expanded?
18:57:54	dom96	yep, that's precisely what happens
18:58:02	dom96	There is no easy solution to your issue I'm afraid
18:58:15	dom96	Jester's macro cannot see into the template
18:58:47	PMunch	But why is transformRouteBody the procedure adding the return?
19:04:04	PMunch	If that was moved to the resp template it would work right?
19:13:52	dom96	feel free to try it, but I'm certain I had a reason for doing it this way. I can't remember what that reason is though :)
19:16:43	PMunch	I'm guessing it was to avoid code repetition
19:16:55	PMunch	return true for all redirect, halt, and resp templates?
19:24:52	PMunch	I'm guessing there's no way in Nim to check if it is a template symbol and then expand it manually in a macro?
19:27:29	FromGitter	<Varriount> PMunch: You might be able to use something with type and/or gensym
19:27:50	FromGitter	<Varriount> Or just use `try` and getAst
19:31:58	PMunch	getAst throws something?
19:32:18	PMunch	dom96, could this work?
19:35:04	PMunch	Oh and by the way: https://github.com/nim-lang/Nim/pull/5778
20:16:09	dom96	I just found out the reason and it's not to do with repetition.
20:16:20	dom96	The body of a route gets put into an async proc
20:16:47	dom96	So if the 'return' is inside the template then the 'async' macro cannot transform this 'return' so we're back to the same issue :)
20:19:04	PMunch	dom96, so templates are broken in async macros as well?
20:19:15	dom96	yep
20:19:19	dom96	you can't have 'await' in a template
20:20:17	PMunch	Hmm, that's not good
20:20:33	dom96	This is more of a general Nim issue
```

**Comment by PMunch** [comment]
Would it be possible to wrap all routes in a block as a solution? So the above would be something like:
```
block route:
  tryWithDbError:
    db.exec(sql"DROP TABLE bobbytables")
    resp Http200, "Dropped bobbytables"
  resp Http404, "This part of the code should never be reached"
```
And instead of returning the resp would do a `break route` and then any returning should be done outside the route block.

**Comment by PMunch** [comment]
Created a PR in  #127 that does what I proposed so that templates and macros can be used in Jester

**Comment by PMunch** [comment]
Fixed in #127
